packages/hoppscotch-sh-admin/src/composables/useConfigHandler.ts

```diff
@@ -212,8 +149,57 @@ export function useConfigHandler(updatedConfigs?: ServerConfigs) {
     );
   };
 
+  /**
+   * The updated configs are transformed into a format that can be used by the mutations
+   */
+
+  const workingUpdatedConfigs: ConfigTransform[] = [
+    {
+      config: GOOGLE_CONFIGS,
+      enabled: updatedConfigs?.providers.google.enabled,
+      field: updatedConfigs?.providers.google.fields,
+    },
+    {
+      config: GITHUB_CONFIGS,
+      enabled: updatedConfigs?.providers.github.enabled,
+      field: updatedConfigs?.providers.github.fields,
+    },
+    {
+      config: MICROSOFT_CONFIGS,
+      enabled: updatedConfigs?.providers.microsoft.enabled,
+      field: updatedConfigs?.providers.microsoft.fields,
+    },
+    {
+      config: MAIL_CONFIGS,
+      enabled: updatedConfigs?.mailConfigs.enabled,
+      field: updatedConfigs?.mailConfigs.fields,
+    },
+  ];
+
+  // Push or filter the configs based on the enabled condition
+  const transformInfraConfigs = (workingConfigs: ConfigTransform[]) => {
+    let newConfigs: UpdatedConfigs[] = [];
+
+    workingConfigs.forEach(({ config, enabled, field }) => {
+      config.forEach(({ name, key }) => {
+        if (enabled && field) {
+          const value = typeof field === 'string' ? field : String(field[key]);
+          newConfigs.push({ name, value });
+        } else {
+          newConfigs = newConfigs.filter((item) => item.name !== name);
+        }
+      });
+    });
+    newConfigs = newConfigs.filter((item) => item.name !== '');
```

Comment:
Aren't entries corresponding to `name` from the config list [maintained statically](https://github.com/hoppscotch/hoppscotch/pull/3971/files/1f8d3e1afbde2dc243aa9811e88fe75c2d172675..957a3ee848e48fd82456eaa3abb97ede7496902b#diff-5403ef86c6b0ecea0765b06a22c80b6620f8d0ddcdd93ef814eb0b0fefdb1b22)? Is there a case where there can be an empty entry, as from the above?